jb/20210103/ubi_spending_children_preprocessing.py

```diff
@@ -66,47 +54,36 @@ def ubi(funding_billions=0, percent=0):
   taking into account that funding will be raise by a flat tax leveled on each households
   taxable income """
 
-  percent = percent / 100
+  child_percent /= 100
 
   funding = funding_billions * 1e9
 
   target_persons = person.copy(deep=True)
 
   # i think this is % funding, not % benefit
-  adult_ubi = ((1 - percent) * funding) / adult_pop
-  child_ubi = (percent * funding) / child_pop
+  adult_ubi = ((1 - child_percent) * funding) / adult_pop
+  child_ubi = (child_percent * funding) / child_pop
 
   tax_rate = funding / total_taxable_income
 
-  target_persons['hh_new_tax'] = target_persons.hh_tax_income * tax_rate
-
-  target_persons['hh_ubi'] = (target_persons.hh_adults * adult_ubi + 
-                              target_persons.hh_children * child_ubi)
-  
-  target_persons['new_spm_resources'] = (target_persons.spm_resources + 
-                                         target_persons.hh_ubi -
-                                         target_persons.hh_new_tax)
-  
-  target_persons['new_spm_resources_pp'] = (target_persons.new_spm_resources / 
-                                            (target_persons.hh_total_people))
+  spmu['new_tax'] = tax_rate * spmu.tax_inc
+  spmu['spm_ubi'] = (spmu.child * child_ubi) + (spmu.adult * adult_ubi)
 
-  
-  # Calculate poverty rate
-  target_persons['poor'] = (target_persons.new_spm_resources < 
-                            target_persons.spm_povthreshold)
-  
-  total_poor = (target_persons.poor * target_persons.weight).sum()
-  poverty_rate = (total_poor / pop * 100)
+  spmu['new_spm_resources'] = spmu.spm_resources + spm_ubi - spmu.new_tax
+  spmu['new_spm_resources_pp'] = spmu.new_spm_resources / spmu.numper
 
   # Calculate poverty gap
-  target_persons['poverty_gap'] = target_persons.spm_povthreshold - target_persons.new_spm_resources
-  spmu=target_persons.drop_duplicates(subset='spm_id')
+  poverty_gap = poverty_gap(spmu, 'new_spm_resources', 'spm_povthreshold', 'spmweight')
 
-  poverty_gap = (((spmu.poor * spmu.poverty_gap
-                             * spmu.weight).sum()))
+  # Merge person and spmu dataframes
+
+  target_persons = person.merge(spmu, left_on='spm_id')
+
+  total_poor = (target_persons.poor * target_persons.person_weight).sum()
```

Comment:
```suggestion
  poverty_rate = mdf.weighted_mean(target_persons, "poor", "person_weight")
```
(and nix next line, and do the same thing with `percent_better_off`. should also be able to nix the `pop` definition above)